metagraph/src/annotation/row_diff_builder.cpp

```diff
@@ -331,38 +333,26 @@ rd_succ_bv_type route_at_forks(const graph::DeBruijnGraph &graph,
             exit(1);
         }
 
-        rd_succ = rd_succ_bv_type(std::move(rd_succ_bv));
-
+        rd_succ = std::make_shared<rd_succ_bv_type>(std::move(rd_succ_bv));
+        rd_succ->serialize(f);
+ 
     } else {
         logger->warn("No count vectors could be found in {}. The last outgoing"
                      " edges will be selected for assigning RowDiff successors",
                      count_vectors_dir);
+        rd_succ = last;
+        if(dynamic_cast<graph::DBGSuccinct const*>(&graph)) {
+            rd_succ_bv_type().serialize(f);
+        } else {
+            rd_succ_bv_type(*rd_succ).serialize(f);
```

Comment:
Not sure what's the best way to avoid copy-construction here, and if it's generally possible for typical results of get_last.